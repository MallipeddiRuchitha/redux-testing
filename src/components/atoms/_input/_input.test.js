import React from 'react';
import { create } from "react-test-renderer";
import { render ,fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Input from './_input';

describe("Input component",() =>{
    test("Matches the snapshot",() => {
        const inputField = create(<Input />);
        expect(inputField.toJSON()).toMatchSnapshot();

    })
    test("input element is rendered",() => {
        const { getByTestId, getByText } = render(<Input />);
        expect(getByTestId("input")).toBeInTheDocument();
        
    });
    test("on change of input",() => {
        const mockOnChange = jest.fn();
        const { getByTestId} = render(<Input value="name"  onChange={mockOnChange}/>);        
        fireEvent.change(getByTestId("value"),{ target: { value: 'ram' } });
        
        expect(mockOnChange).toHaveBeenCalledTimes(1);
    });
   

});