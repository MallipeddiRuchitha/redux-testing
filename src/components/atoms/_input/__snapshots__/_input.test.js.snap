// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleInputFields-1"
  data-testid="input"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline"
    onClick={[Function]}
    required={true}
    style={
      Object {
        "color": "#000",
      }
    }
  >
    <input
      className="MuiInputBase-input MuiInput-input"
      data-testid="value"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
    />
  </div>
</div>
`;
